src/Components/Facility/Consultations/DailyRoundsFilter.tsx

```diff
@@ -33,14 +33,31 @@ export default function DailyRoundsFilter(props: Props) {
     errorClassName: "hidden",
   });
 
+  const countDefinedProperties = (obj) => {
+    return Object.values(obj).filter((val) => val !== undefined).length;
+  };
```

Comment:
Why do an iterate and check for length? You can simply use the `Array.every`/`Array.some` method for this instead, which would be much more readable.

```suggestion
  const isFilterApplied = Object.values(filter ?? {}).some(Boolean);
```